backend/hitas/models/owner.py

```diff
@@ -38,40 +38,39 @@ def save(self, *args, **kwargs) -> None:
         super().save(*args, **kwargs)
 
     @classmethod
-    def post_fetch_hook(cls, results, fields):
-        for i, owner in enumerate(results):
-            # qs.values(...)
-            if isinstance(owner, dict):
-                # If non_disclosure is not given, we assume the owner needs to be obfuscated
-                if "non_disclosure" not in owner or owner["non_disclosure"] is True:
-                    for field in cls.obfuscation_rules:
-                        if field in owner:
-                            owner[field] = None
-
-            # qs.values_list(...)
-            elif isinstance(owner, tuple):
-                # Fields that were accessed
-                index_nd = index_of(fields, "non_disclosure")
-
-                # If non_disclosure is not given, we assume the owner needs to be obfuscated
-                if index_nd is None or owner[index_nd] is True:
-                    for field in cls.obfuscation_rules:
-                        field_index = index_of(fields, field)
-                        if field_index is not None:
-                            results[i] = (*owner[:field_index], None, *owner[field_index + 1 :])
-
-            # qs.get(...), qs.all(), etc.
-            elif isinstance(owner, Owner):
-                obfuscate(owner)
-
-            # qs.values_list(..., flat=True)
-            else:
-                # We cannot know if the owner should be obfuscated or not, so we assume it should be.
-                # This should probably not be used, but it's here just in case.
-                if fields[0] in cls.obfuscation_rules:
-                    results[i] = None
-
-        return results
+    def post_fetch_hook(cls, model: "Owner") -> "Owner":
+        obfuscate(model)
+        return model
+
+    @classmethod
+    def post_fetch_values_hook(cls, values: dict[str, Any], fields: tuple[str, ...]) -> dict[str, Any]:
+        # If non_disclosure is not given, we assume the owner needs to be obfuscated
+        if "non_disclosure" not in values or values["non_disclosure"] is True:
+            for field, value in cls.obfuscation_rules.items():
+                if field in values:
+                    values[field] = value
+        return values
+
+    @classmethod
+    def post_fetch_values_list_hook(cls, values: tuple[Any, ...], fields: tuple[str, ...]) -> tuple[Any, ...]:
+        # Fields that were accessed
+        index_nd = index_of(fields, "non_disclosure")
+
+        # If non_disclosure is not given, we assume the owner needs to be obfuscated
```

Comment:
I'm not a fan of *assuming* that everything needs to be obfuscated if `non_disclosure` field is missing, instead i'd rather:
- Raise an exception or 
- Add the `non_disclosure` field to the values, and later pop it out

Reply:
I do the add-and-pop method already here: 

https://github.com/City-of-Helsinki/hitas/blob/39c7f5add9205fe306b4281e01ecb1546f3861f7/backend/hitas/services/owner.py#L39 

Maybe making it mandatory to include by raising a RuntimeExcepition could be a better way?

Reply:
Hmm yes, then we would need to do less 'hacky' stuff 😄

Reply:
Done.